File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> m = new HashMap<>();
        for(String k : participant){
            m.put(k, m.getOrDefault(k, 0) + 1);
        }
        
        for(String k : completion){
            m.put(k, m.get(k) - 1);
        }
        
        for(String k : m.keySet()){
            if(m.get(k) != 0){
                return k;
            }
        }
        
        return "";
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> count;
    
    for(string p : participant){
        count[p]++;
    }
    
    for(string c : completion){
        count[c]--;
    }
    
    for(auto a : count){
        if(a.second > 0){
            return a.first;
        }
    }
    
    return "";
}